scripts/verification.py

```diff
@@ -73,29 +91,37 @@
     nl_code = -1
 
 # End the final newline tests
-# ---------------------------
+# --------------------------------------
+# Reload file since it has been consumed
```

Comment:
As an alternative to reopening the file multiple times (which is expensive IO) you could change the utility functions so that they take strings rather than file handles. Then the file only needs to be read once. As an intermediate step, take a look at rewinding the file position (though I'd recommend the string approach).

Reply:
Done that with 7e1ed3aaa92a991e64e133448109cd17d73e2d60.